app/controllers/ops_controller.rb

```diff
@@ -716,11 +722,29 @@ def rbac_replace_right_cell(nodetype, presenter)
       @right_cell_text = @edit ?
           _("Manage quotas for %{model} \"%{name}\"") % {:name => @tenant.name, :model => model} :
           _("%{model} \"%{name}\"") % {:model => model, :name => @tenant.name}
+    elsif nodetype == 'dialog_return'
+      presenter.update(:main_div, r[:partial => "detail_page"])
     else
       presenter[:update_partials][@sb[:active_tab].to_sym] = r[:partial => "#{@sb[:active_tab]}_tab"]
     end
   end
 
+  # set all needed thing before calling replace_right_cell with nodetype
+  def dialog_replace_right_cell
+    model, id = TreeBuilder.extract_node_model_and_id(x_node)
+    @record = model.constantize.find(from_cid(id))
+    case @record
+    when User
+      @user = @record
+    when MiqGroup
+      @group = @record
+      rbac_group_get_details(@record.id)
+    when Tenant
+      @tenant = @record
+    end
```

Comment:
I'd prefer to remove this case and work with the users of `@sb[:tab_label]`

I'd rather create some heler that returns the `rbac_set_tab_label` and get's called from the users of `@sb[:tab_label]`

From the first `grep` it seems to me that the session should not be needed at all.